held_harp.py

```diff
@@ -0,0 +1,98 @@
+from Algorithms.ShortestPath import ShortestPath
+import json
+
+
+def graph_creation(graph, start, to_visit) -> tuple:
+    stops = [start] + to_visit
+    new_graph = [[0]*len(stops) for _ in range(len(stops))]
+    indexes = []
+    paths = [[0]*len(stops) for _ in range(len(stops))]
+    for i in range(len(stops)):
+        indexes.append([i, stops[i]])  # Each sublist contains [current_index, IdP]
+        for j in range(len(stops)):
+            result = s.dijkstra(graph, stops[i], stops[j])
+            paths[i][j] = result[0]
+            new_graph[i][j] = result[1]
+    return new_graph, indexes, paths
+
+
+def held_karp(graph: list, start: int, to_go: list, reach: int) -> list:
```

Comment:
Spróbuj jakoś tak te ścieżki zapisywać, że na początku masz pustą listę, potem do niej robisz append kolejnych elementów tak jak w tym kodzie dodają się symbole kodu w kolejnych iteracjach. Warto wziąć też pod uwagę, że rekursja jest procesem czasochłonnym, więc złożoność może być nie taka. 

def encode(self, node: HuffmanTreeNode, letter: str, code: list) -> list:
        """
        Returns letter code in Huffman Compression Tree. Initialize with code = [].
        """
        if node:
            if letter == node.letter:
                return code
            if letter in node.left.letter:
                code.append(0)
                return self.encode(node.left, letter, code)
            else:
                code.append(1)
                return self.encode(node.right, letter, code)
        else:
            return code